client/my-sites/plugins/plans/index.tsx

```diff
@@ -14,18 +18,27 @@ import FixedNavigationHeader from 'calypso/components/fixed-navigation-header';
 import FormattedHeader from 'calypso/components/formatted-header';
 import MainComponent from 'calypso/components/main';
 import PromoSection, { Props as PromoSectionProps } from 'calypso/components/promo-section';
+import { useWPCOMPlugin } from 'calypso/data/marketplace/use-wpcom-plugins-query';
 import { Gridicon } from 'calypso/devdocs/design/playground-scope';
 import PageViewTracker from 'calypso/lib/analytics/page-view-tracker';
 import PlansFeaturesMain from 'calypso/my-sites/plans-features-main';
 import { MarketplaceFooter } from 'calypso/my-sites/plugins/education-footer';
 import { MARKETPLACE_FLOW } from 'calypso/my-sites/plugins/flows';
 import { appendBreadcrumb } from 'calypso/state/breadcrumb/actions';
 import { getBreadcrumbs } from 'calypso/state/breadcrumb/selectors';
+import { getProductsList } from 'calypso/state/products-list/selectors';
 import { getSelectedSite } from 'calypso/state/ui/selectors';
+import CTAButton from '../plugin-details-CTA/CTA-button';
 
 import './style.scss';
 
-const Plans = ( { intervalType }: { intervalType: 'yearly' | 'monthly' } ) => {
+const Plans = ( {
+	intervalType,
+	pluginSlug,
+}: {
+	intervalType: 'yearly' | 'monthly';
+	pluginSlug: string;
```

Comment:
```suggestion
	pluginSlug?: string;
```

or.. actually can we just create two plans components, one for when the plugin param is present, one for when it's not.